src/sqltest/services/connectionManagement/connectionManagementService.test.ts

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as TypeMoq from 'typemoq';
+import { ConnectionManagementService } from 'sql/platform/connection/common/connectionManagementService';
+import { ConnectionStatusManager } from 'sql/platform/connection/common/connectionStatusManager';
+import { ConnectionStore } from 'sql/platform/connection/common/connectionStore';
+import { TestStorageService } from 'vs/workbench/test/workbenchTestServices';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { CapabilitiesTestService } from 'sqltest/stubs/capabilitiesTestService';
+import { ConnectionOptionSpecialType, ServiceOptionType } from 'sql/workbench/api/common/sqlExtHostTypes';
+import { ConnectionProfileGroup } from 'sql/platform/connection/common/connectionProfileGroup';
+
+const sqlProvider = {
```

Comment:
>sqlProvider [](start = 6, length = 11)

It looks like the CapabilitiesTestService already defines a test provider?

If there's a reason you have a special one here I'd suggest at least pulling this into its own file - it's something I'm sure other tests would want to use.

Reply:
good catch